packages/iov-stream/src/defaultvalueproducer.ts

```diff
@@ -0,0 +1,63 @@
+import { Listener, Producer } from "xstream";
+
+export interface DefaultValueProducerCallsbacks {
+  readonly onStarted: () => void;
+  readonly onStop: () => void;
+}
+
+// allows pre-producing values before anyone is listening
+export class DefaultValueProducer<T> implements Producer<T> {
```

Comment:
This is just moving it into it's own file and a bit of doc cleanup, right?

Reply:
The most impotant change is the addition of the `onStarted` and `onStop` callbacks, that did not exist before. Before we used DefaultValueProducer in cases where producing updates was cheap and we produced no matter if anyone is listening or not. Now, producing updates requires polling, so we can enable/disable polling only when someone cares for the updates.